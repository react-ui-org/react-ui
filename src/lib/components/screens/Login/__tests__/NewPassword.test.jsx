import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import { NewPassword } from '..';

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = mount(<NewPassword />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with all props except translations', () => {
    const tree = mount(<NewPassword
      id="custom-id"
      logo="http://satyr.io/100x100/33?text=logo"
      title="Company"
      error="Error message"
      footer={(
        <a href="http://example.com">link</a>
      )}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with translations', () => {
    const tree = mount(<NewPassword
      translations={{
        changePassword: 'Změnit heslo',
        newPassword: 'Nové heslo',
        repeatNewPassword: 'Nové heslo znovu',
      }}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls submitHandler() on submit', () => {
    const spy = sinon.spy();
    const component = mount(<NewPassword submitHandler={spy} />);

    component
      .find('form')
      .simulate('submit');
    expect(spy.calledOnce).toEqual(true);
  });
});
