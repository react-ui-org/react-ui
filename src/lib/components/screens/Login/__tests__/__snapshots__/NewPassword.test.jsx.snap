// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<WithTranslationContextComponent
  translations={null}
>
  <NewPassword
    error={null}
    footer={null}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "changePassword": "Change password",
        "newPassword": "New password",
        "repeatNewPassword": "Repeat new password",
      }
    }
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <TextField
              autoComplete="new-password"
              changeHandler={[Function]}
              disabled={false}
              fullWidth={true}
              helperText={null}
              id="newPasswordInput"
              inputSize={null}
              isLabelVisible={true}
              label="New password"
              layout="vertical"
              placeholder={null}
              required={true}
              size="medium"
              type="password"
              validationState={null}
              variant="outline"
            >
              <label
                className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        rootVariantOutline"
                htmlFor="newPasswordInput"
              >
                <div
                  className="label"
                  id="newPasswordInput__label"
                >
                  New password
                </div>
                <div
                  className="inputContainer"
                >
                  <input
                    autoComplete="new-password"
                    className="input"
                    disabled={false}
                    id="newPasswordInput"
                    onChange={[Function]}
                    placeholder={null}
                    required={true}
                    size={null}
                    type="password"
                  />
                </div>
              </label>
            </TextField>
            <TextField
              autoComplete="new-password"
              changeHandler={[Function]}
              disabled={false}
              fullWidth={true}
              helperText={null}
              id="newPasswordRepeatInput"
              inputSize={null}
              isLabelVisible={true}
              label="Repeat new password"
              layout="vertical"
              placeholder={null}
              required={true}
              size="medium"
              type="password"
              validationState={null}
              variant="outline"
            >
              <label
                className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        rootVariantOutline"
                htmlFor="newPasswordRepeatInput"
              >
                <div
                  className="label"
                  id="newPasswordRepeatInput__label"
                >
                  Repeat new password
                </div>
                <div
                  className="inputContainer"
                >
                  <input
                    autoComplete="new-password"
                    className="input"
                    disabled={false}
                    id="newPasswordRepeatInput"
                    onChange={[Function]}
                    placeholder={null}
                    required={true}
                    size={null}
                    type="password"
                  />
                </div>
              </label>
            </TextField>
          </div>
          <Button
            afterLabel={null}
            beforeLabel={null}
            block={true}
            clickHandler={null}
            disabled={false}
            endCorner={null}
            grouped={false}
            label="Change password"
            labelVisibility="all"
            loadingIcon={null}
            priority="default"
            size="medium"
            startCorner={null}
            type="submit"
            variant="primary"
          >
            <button
              className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
              disabled={false}
              onClick={null}
              title={null}
              type="submit"
            >
              <span
                className="label"
              >
                Change password
              </span>
            </button>
          </Button>
        </form>
      </div>
    </div>
  </NewPassword>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with all props except translations 1`] = `
<WithTranslationContextComponent
  error="Error message"
  footer={
    <a
      href="http://example.com"
    >
      link
    </a>
  }
  id="custom-id"
  logo="http://satyr.io/100x100/33?text=logo"
  title="Company"
  translations={null}
>
  <NewPassword
    error="Error message"
    footer={
      <a
        href="http://example.com"
      >
        link
      </a>
    }
    id="custom-id"
    logo="http://satyr.io/100x100/33?text=logo"
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title="Company"
    translations={
      Object {
        "changePassword": "Change password",
        "newPassword": "New password",
        "repeatNewPassword": "Repeat new password",
      }
    }
  >
    <div
      className="root"
      id="custom-id"
    >
      <div
        className="title"
        id="custom-id__title"
      >
        Company
      </div>
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="error"
            id="custom-id__errorText"
          >
            Error message
          </div>
          <div
            className="mb-3"
          >
            <TextField
              autoComplete="new-password"
              changeHandler={[Function]}
              disabled={false}
              fullWidth={true}
              helperText={null}
              id="custom-id__newPasswordInput"
              inputSize={null}
              isLabelVisible={true}
              label="New password"
              layout="vertical"
              placeholder={null}
              required={true}
              size="medium"
              type="password"
              validationState={null}
              variant="outline"
            >
              <label
                className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        rootVariantOutline"
                htmlFor="custom-id__newPasswordInput"
              >
                <div
                  className="label"
                  id="custom-id__newPasswordInput__label"
                >
                  New password
                </div>
                <div
                  className="inputContainer"
                >
                  <input
                    autoComplete="new-password"
                    className="input"
                    disabled={false}
                    id="custom-id__newPasswordInput"
                    onChange={[Function]}
                    placeholder={null}
                    required={true}
                    size={null}
                    type="password"
                  />
                </div>
              </label>
            </TextField>
            <TextField
              autoComplete="new-password"
              changeHandler={[Function]}
              disabled={false}
              fullWidth={true}
              helperText={null}
              id="custom-id__newPasswordRepeatInput"
              inputSize={null}
              isLabelVisible={true}
              label="Repeat new password"
              layout="vertical"
              placeholder={null}
              required={true}
              size="medium"
              type="password"
              validationState={null}
              variant="outline"
            >
              <label
                className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        rootVariantOutline"
                htmlFor="custom-id__newPasswordRepeatInput"
              >
                <div
                  className="label"
                  id="custom-id__newPasswordRepeatInput__label"
                >
                  Repeat new password
                </div>
                <div
                  className="inputContainer"
                >
                  <input
                    autoComplete="new-password"
                    className="input"
                    disabled={false}
                    id="custom-id__newPasswordRepeatInput"
                    onChange={[Function]}
                    placeholder={null}
                    required={true}
                    size={null}
                    type="password"
                  />
                </div>
              </label>
            </TextField>
          </div>
          <Button
            afterLabel={null}
            beforeLabel={null}
            block={true}
            clickHandler={null}
            disabled={false}
            endCorner={null}
            grouped={false}
            id="custom-id__changePasswordButton"
            label="Change password"
            labelVisibility="all"
            loadingIcon={null}
            priority="default"
            size="medium"
            startCorner={null}
            type="submit"
            variant="primary"
          >
            <button
              className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
              disabled={false}
              id="custom-id__changePasswordButton"
              onClick={null}
              title={null}
              type="submit"
            >
              <span
                className="label"
                id="custom-id__changePasswordButton__label"
              >
                Change password
              </span>
            </button>
          </Button>
        </form>
        <div
          className="footer"
          id="custom-id__footerContent"
        >
          <a
            href="http://example.com"
          >
            link
          </a>
        </div>
      </div>
    </div>
  </NewPassword>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with translations 1`] = `
<WithTranslationContextComponent
  translations={
    Object {
      "changePassword": "Změnit heslo",
      "newPassword": "Nové heslo",
      "repeatNewPassword": "Nové heslo znovu",
    }
  }
>
  <NewPassword
    error={null}
    footer={null}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "changePassword": "Změnit heslo",
        "newPassword": "Nové heslo",
        "repeatNewPassword": "Nové heslo znovu",
      }
    }
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <TextField
              autoComplete="new-password"
              changeHandler={[Function]}
              disabled={false}
              fullWidth={true}
              helperText={null}
              id="newPasswordInput"
              inputSize={null}
              isLabelVisible={true}
              label="Nové heslo"
              layout="vertical"
              placeholder={null}
              required={true}
              size="medium"
              type="password"
              validationState={null}
              variant="outline"
            >
              <label
                className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        rootVariantOutline"
                htmlFor="newPasswordInput"
              >
                <div
                  className="label"
                  id="newPasswordInput__label"
                >
                  Nové heslo
                </div>
                <div
                  className="inputContainer"
                >
                  <input
                    autoComplete="new-password"
                    className="input"
                    disabled={false}
                    id="newPasswordInput"
                    onChange={[Function]}
                    placeholder={null}
                    required={true}
                    size={null}
                    type="password"
                  />
                </div>
              </label>
            </TextField>
            <TextField
              autoComplete="new-password"
              changeHandler={[Function]}
              disabled={false}
              fullWidth={true}
              helperText={null}
              id="newPasswordRepeatInput"
              inputSize={null}
              isLabelVisible={true}
              label="Nové heslo znovu"
              layout="vertical"
              placeholder={null}
              required={true}
              size="medium"
              type="password"
              validationState={null}
              variant="outline"
            >
              <label
                className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        rootVariantOutline"
                htmlFor="newPasswordRepeatInput"
              >
                <div
                  className="label"
                  id="newPasswordRepeatInput__label"
                >
                  Nové heslo znovu
                </div>
                <div
                  className="inputContainer"
                >
                  <input
                    autoComplete="new-password"
                    className="input"
                    disabled={false}
                    id="newPasswordRepeatInput"
                    onChange={[Function]}
                    placeholder={null}
                    required={true}
                    size={null}
                    type="password"
                  />
                </div>
              </label>
            </TextField>
          </div>
          <Button
            afterLabel={null}
            beforeLabel={null}
            block={true}
            clickHandler={null}
            disabled={false}
            endCorner={null}
            grouped={false}
            label="Změnit heslo"
            labelVisibility="all"
            loadingIcon={null}
            priority="default"
            size="medium"
            startCorner={null}
            type="submit"
            variant="primary"
          >
            <button
              className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
              disabled={false}
              onClick={null}
              title={null}
              type="submit"
            >
              <span
                className="label"
              >
                Změnit heslo
              </span>
            </button>
          </Button>
        </form>
      </div>
    </div>
  </NewPassword>
</WithTranslationContextComponent>
`;
